client/go/outline/fetch.go

```diff
@@ -17,16 +17,22 @@ package outline
 import (
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
 )
 
+const fetchTimeout = 10 * time.Second
```

Comment:
Ideally the TS code would specify this, but this is a good mitigation.

Reply:
Ideally, TS code should combine `fetchResource` and the next `establishVPN` into a single "transaction", so both operations can be cancelled by a single `context` object.